File: update_images.py

import json
import os

file_path = r'c:\Users\Benson\Desktop\cooker\src\data\ingredients_master.json'

def kebab_to_title(text):
    return ' '.join(word.capitalize() for word in text.split('-'))

print(f"Reading {file_path}...")
with open(file_path, 'r', encoding='utf-8') as f:
    ingredients = json.load(f)

updated_count = 0
for ingredient in ingredients:
    # Convert ID to Title Case
    # Special fix for some known discrepancies if needed, but per instructions we do algorithmic map first
    # e.g. "purple-rice" -> "Purple Rice"
    
    # Check if we should skip manual logic? No, update ALL per instructions.
    
    raw_id = ingredient.get('id', '')
    if raw_id:
        title_name = kebab_to_title(raw_id)
        
        # Specific TheMealDB naming conventions often use specific spellings
        # But we start with the requested rule.
        
        new_url = f"https://www.themealdb.com/images/ingredients/{title_name}.png"
        ingredient['imageUrl'] = new_url
        updated_count += 1

print(f"Updated {updated_count} ingredients.")

with open(file_path, 'w', encoding='utf-8') as f:
    json.dump(ingredients, f, indent=4, ensure_ascii=False)

print("Done.")

File: update_images.js

const fs = require('fs');
const path = require('path');

const filePath = path.join('c:', 'Users', 'Benson', 'Desktop', 'cooker', 'src', 'data', 'ingredients_master.json');

console.log(`Reading ${filePath}...`);
try {
    const data = fs.readFileSync(filePath, 'utf8');
    const ingredients = JSON.parse(data);

    let updatedCount = 0;
    ingredients.forEach(ingredient => {
        if (ingredient.id) {
            // Convert kebab-case to Title Case
            const titleName = ingredient.id
                .split('-')
                .map(word => word.charAt(0).toUpperCase() + word.slice(1))
                .join(' ');

            ingredient.imageUrl = `https://www.themealdb.com/images/ingredients/${titleName}.png`;
            updatedCount++;
        }
    });

    console.log(`Updated ${updatedCount} ingredients.`);
    fs.writeFileSync(filePath, JSON.stringify(ingredients, null, 4), 'utf8');
    console.log("Done.");

} catch (err) {
    console.error("Error:", err);
}
